node/src/services/builder.rs

```diff
@@ -0,0 +1,166 @@
+use async_channel::Receiver;
+use sc_network::{config::IncomingRequest, ProtocolName};
+use sc_service::RpcHandlers;
+use shc_common::types::HasherOutT;
+use sp_keystore::KeystorePtr;
+use sp_trie::TrieLayout;
+use std::{marker::PhantomData, sync::Arc};
+use tokio::sync::RwLock;
+
+use file_manager::{in_memory::InMemoryFileStorage, traits::FileStorage};
+use forest_manager::{
+    in_memory::InMemoryForestStorage, rocksdb::RocksDBForestStorage, traits::ForestStorage,
+};
+use storage_hub_infra::actor::{ActorHandle, TaskSpawner};
+
+use crate::service::{ParachainClient, ParachainNetworkService};
+
+use super::{
+    blockchain::{spawn_blockchain_service, BlockchainService},
+    file_transfer::{spawn_file_transfer_service, FileTransferService},
+    handler::StorageHubHandler,
+};
+
+/// Builds the `StorageHubHandler` by adding each component separately.
+/// Provides setters and getters for each component.
+pub struct StorageHubBuilder<T, FL, FS> {
+    task_spawner: Option<TaskSpawner>,
+    file_transfer: Option<ActorHandle<FileTransferService>>,
+    blockchain: Option<ActorHandle<BlockchainService>>,
+    file_storage: Option<Arc<RwLock<FL>>>,
+    forest_storage: Option<Arc<RwLock<FS>>>,
+    provider_pub_key: Option<[u8; 32]>,
+    _marker: PhantomData<T>,
+}
+
+impl<T, FL, FS> StorageHubBuilder<T, FL, FS>
+where
+    T: TrieLayout + Send + Sync + 'static,
+    FL: FileStorage<T> + Send + Sync,
+    FS: ForestStorage<T> + Send + Sync + 'static,
+    HasherOutT<T>: TryFrom<[u8; 32]>,
+{
+    pub fn new(task_spawner: TaskSpawner) -> Self {
+        Self {
+            task_spawner: Some(task_spawner),
+            file_transfer: None,
+            blockchain: None,
+            file_storage: None,
+            forest_storage: None,
+            provider_pub_key: None,
+            _marker: Default::default(),
+        }
+    }
+
+    pub async fn with_file_transfer(
```

Comment:
I would add some small docs for each of the `with_*` functions just to give some clarity on what this would actually be doing in the background.